lib/ansible/modules/cloud/amazon/iam_policy.py

```diff
@@ -112,240 +114,215 @@
         policy_json=dict(type='json', default=None, required=False),
         skip_duplicates=dict(type='bool', default=True, required=False)
-    ))
-
-    module = AnsibleModule(
-        argument_spec=argument_spec,
     )
-
-    if not HAS_BOTO:
-        module.fail_json(msg='boto required for this module')
-
-    iam_type = module.params.get('iam_type').lower()
-    state = module.params.get('state')
-    name = module.params.get('iam_name')
-    policy_name = module.params.get('policy_name')
-    skip = module.params.get('skip_duplicates')
-
-    policy_document = module.params.get('policy_document')
-    if policy_document is not None and module.params.get('policy_json') is not None:
-        module.fail_json(msg='Only one of "policy_document" or "policy_json" may be set')
-
-    if policy_document is not None:
-        try:
-            with open(policy_document, 'r') as json_data:
-                pdoc = json.dumps(json.load(json_data))
-                json_data.close()
-        except IOError as e:
-            if e.errno == 2:
-                module.fail_json(
-                    msg='policy_document {0:!r} does not exist'.format(policy_document))
-            else:
-                raise
-    elif module.params.get('policy_json') is not None:
-        pdoc = module.params.get('policy_json')
-        # if its a string, assume it is already JSON
-        if not isinstance(pdoc, string_types):
-            try:
-                pdoc = json.dumps(pdoc)
-            except Exception as e:
-                module.fail_json(msg='Failed to convert the policy into valid JSON: %s' % str(e))
-    else:
-        pdoc = None
-
-    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module)
+    mutually_exclusive = [['policy_document', 'policy_json']]
+
+    module = AnsibleAWSModule(argument_spec=argument_spec, mutually_exclusive=mutually_exclusive, supports_check_mode=True)
+
+    args = dict(
+        client=module.client('iam'),
+        name=module.params.get('iam_name'),
+        policy_name=module.params.get('policy_name'),
+        policy_document=module.params.get('policy_document'),
+        policy_json=module.params.get('policy_json'),
+        skip_duplicates=module.params.get('skip_duplicates'),
+        state=module.params.get('state'),
+        check_mode=module.check_mode,
+    )
+    iam_type = module.params.get('iam_type')
 
     try:
-        if region:
-            iam = connect_to_aws(boto.iam, region, **aws_connect_kwargs)
-        else:
-            iam = boto.iam.connection.IAMConnection(**aws_connect_kwargs)
-    except boto.exception.NoAuthHandlerFound as e:
+        if iam_type == 'user':
+            policy = UserPolicy(**args)
+        elif iam_type == 'role':
+            policy = RolePolicy(**args)
+        elif iam_type == 'group':
+            policy = GroupPolicy(**args)
+
+        module.exit_json(**(policy.run()))
+    except (BotoCoreError, ClientError, ParamValidationError) as e:
```

Comment:
ParamValidationError is caught already by BotoCoreError, so this can just be
```suggestion
    except (BotoCoreError, ClientError) as e:
```
You can also remove the import for ParamValidationError